### PR TITLE
Fix BSA 181 - Fix error on paste with ckeditor relative to upgrade to new version of that library

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to this issue**

https://oat-sa.atlassian.net/browse/BSA-181

**Description**

When the candidate answers an open-ended question, it is not possible to copy using the toolbar button. An error message appears inside the text block. The same behavior is identified during the work with the same item in the item authoring mode.

**Preconditions**

- Created test, which includes the item with Extended text common interaction with set XHTML Format in the properties;

**Steps**

- Start the delivery;
- Open the item with Extended text interaction and enter the text;
- Cope the text by clicking the Copy button in the tools bar;
- Try to paste the copied text by the Paste button from the tools bar;

**Actual behavior**

The text is not pasted; there is a warning message, but it appeared inside the text block. not in the pop-up or modal window. 

**Expected behavior**

The system should enable the test taker to copy and paste a word or a phrase in the content area of the open question.

image

### CODE REVIEW

File: lib/ckeditor/config.js [hunk 1]
```diff
@@ -9,3 +9,15 @@ CKEDITOR.editorConfig = function (config) {
...
 };
 
+CKEDITOR.on('instanceReady', function(event) {
+    event.editor.on('beforeCommandExec', function(event) {
+        // Show the paste dialog for the paste buttons and right-click paste
+        if (event.data.name == 'paste') {
+            event.editor._.forcePasteDialog = true;
+        }
+        // Don't show the paste dialog for Ctrl+Shift+V
+        if (event.data.name == 'pastetext' && event.data.commandData.from == 'keystrokeHandler') {
```
Comment (wazelin) [inline lib/ckeditor/config.js:19]:
```suggestion
        // Show the paste dialog for the paste buttons and context-menu paste
        if (event.data.name == 'paste') {
            event.editor._.forcePasteDialog = true;

            return;
        }

        // Don't show the paste dialog for Ctrl/Cmd + Shift + V
        if (event.data.name === 'pastetext' && event.data.commandData.from === 'keystrokeHandler') {

```

File: lib/ckeditor/config.js [hunk 2]
```diff
@@ -9,3 +9,19 @@ CKEDITOR.editorConfig = function (config) {
 	// config.uiColor = '#AADC6E';
 };
 
+CKEDITOR.on('instanceReady', function(event) {
```
Comment (andrey-ponamarev) [inline lib/ckeditor/config.js:12]: Please add your changes to the https://github.com/oat-sa/ckeditor-dev and follow instruction to generate release https://github.com/oat-sa/ckeditor-dev#build

Comment (habemuscode) [reply lib/ckeditor/config.js:12]: done

**Comment by bziondik** [comment]
In delivery all works 
image

But in item Preview we have errors:
image
image
image

**Comment by habemuscode** [comment]
@andrey-ponamarev Found the solution for @bziondik issue. I update all 3 PRs

File: lib/ckeditor/plugins/dialog/styles/dialog.css [hunk 3]
```diff
@@ -10,7 +10,7 @@
 	height: 100%;
 	top: 0;
 	left: 0;
-	z-index: 10010;
+	z-index: 999999 !important;
```
Comment (andrey-ponamarev) [inline lib/ckeditor/plugins/dialog/styles/dialog.css:13]: Could you use option [baseFloatZIndex](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-baseFloatZIndex) in config instead?

Comment (habemuscode) [reply lib/ckeditor/plugins/dialog/styles/dialog.css]: Done, thanks!

**Comment by habemuscode** [comment]
Changed to core variable to avoid conflicts: https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-baseFloatZIndex  @andrey-ponamarev

**Pull request merged**